Lib/dataclasses.py

```diff
@@ -1299,11 +1299,12 @@ def _add_slots(cls, is_frozen, weakref_slot, defined_fields):
     # Clear existing `__weakref__` descriptor, it belongs to a previous type:
     cls_dict.pop('__weakref__', None)  # gh-102069
 
+    # Set the `__qualname__ accordingly
+    if (qualname := getattr(cls, '__qualname__', None) is not None):
```

Comment:
```suggestion
    if (qualname := getattr(cls, '__qualname__', None)) is not None:
```

Reply:
You know, after I typed it out I felt weird, like I was expecting red squiggles, but didn't give a second thought.

This explains it 😅